spark-3/src/test/scala/org/neo4j/spark/DataSourceReaderTSE.scala

```diff
@@ -1770,6 +1771,37 @@ class DataSourceReaderTSE extends SparkConnectorScalaBaseTSE {
     )
   }
 
+  @Test
+  def testShouldAggregateAndLimitTheRelationshipResults(): Unit = {
+    val total = 100
+    val fixtureQuery: String =
+      s"""UNWIND range(1, $total) as id
+         |CREATE (pr:Product {id: id * rand(), name: 'Product ' + id})
+         |CREATE (pe:Person {id: id, fullName: 'Person ' + id})
+         |CREATE (pe)-[:BOUGHT{when: rand(), quantity: rand() * 1000}]->(pr)
+         |RETURN *
+    """.stripMargin
+
+    SparkConnectorScalaSuiteIT.session()
+      .writeTransaction(
+        (tx: Transaction) => tx.run(fixtureQuery).consume())
+
+    val df = ss.read
+      .format(classOf[DataSource].getName)
+      .option("url", SparkConnectorScalaSuiteIT.server.getBoltUrl)
+      .option("relationship", "BOUGHT")
+      .option("relationship.source.labels", "Person")
+      .option("relationship.target.labels", "Product")
+      .load
+      .select("`target.name`", "`target.id`")
+      .orderBy(col("`target.name`").desc) // FIXME (generated query is wrong)
```

Comment:
is this `FIXME` just forgotten or is there something that we really need to fix?

Reply:
ah wait, I think we forgot about it. There was a test case where it was causing issues